### PR TITLE
EVG-13523 Task annotations analytics

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: src/analytics/task/useTaskAnalytics.ts [hunk 1]
```diff
@@ -40,7 +40,17 @@ type Action =
...
+  | { name: "Save Annotation Note" }
+  | { name: "Click Annotation Ticket Link" }
+  | { name: "Remove Annotation Issue" }
+  | { name: "Remove Annotation Suspected Issue" }
+  | { name: "Move Annotation Issue" }
+  | { name: "Move Annotation Suspected Issue" }
+  | { name: "Click Add Annotation Issue Button" }
+  | { name: "Click Add Annotation Suspected Issue Button" }
+  | { name: "Add Task Annotation Issue" }
+  | { name: "Add Task Annotation Suspected Issue" };
```
Comment (khelif96) [inline src/analytics/task/useTaskAnalytics.ts:53]: Maybe these should live in a separate build baron analytics hook?

File: src/gql/generated/types.ts [hunk 2]
```diff
@@ -1212,12 +1211,76 @@ export type GetTaskEventDataQueryVariables = Exact<{
 export type GetTaskEventDataQuery = {
   task?: Maybe<{
     id: string;
-    execution: number;
+    execution?: Maybe<number>;
```
Comment (khelif96) [inline src/gql/generated/types.ts:1214]: i think the `schema.graphql` you generated this with is out of date this field is always there. Pull the latest evergreen master and run `yarn codegen` again

File: src/analytics/task/query.ts [hunk 3]
```diff
@@ -10,3 +10,74 @@ export const GET_TASK_EVENT_DATA = gql`
     }
   }
 `;
+
+export const GET_ANNOTATION_EVENT_DATA = gql`
```
Comment (khelif96) [inline src/analytics/task/query.ts:14]: Are all these fields necessary for the data needed for the annotation analytics?

File: src/analytics/task/useTaskAnalytics.ts (hunk 1 above)
Comment (khelif96) [inline src/analytics/task/useTaskAnalytics.ts:53]: Thse should be removed from here since they already exist in the useAnnnotationAnalytics hook

File: src/components/BuildBaronAndAnnotations/AddIssueModal.tsx [hunk 4]
```diff
@@ -33,6 +34,7 @@ export const AddIssueModal: React.FC<Props> = ({
   execution,
   isIssue,
 }) => {
+  const taskAnalytics = useTaskAnalytics();
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AddIssueModal.tsx:37]: This should be replaced with the useAnnotationAnalytics hook you created.

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: src/analytics/task/query.ts (hunk 3 above)
Comment (malikchaya2) [reply src/analytics/task/query.ts:14]: I spoke to Maria and we figure out which fields can be removed. Thanks!

**Comment by malikchaya2** [comment]
@khelif96  I accidentally combined my original changes with the pr changes, my apologies. Hopefully it's not too hard to follow.

**Comment by malikchaya2** [comment]
The save note field event can't be tested currently, but @ablack12  will test it after completing [EVG-13815](https://jira.mongodb.org/browse/EVG-13815)

File: src/analytics/task/useAnnotationAnalytics.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      variables: { taskId: id, execution },
+      fetchPolicy: "cache-first",
+    }
+  );
+
+  const annotation = get(eventData, "task.annotation", undefined);
+  const bbConfigured = get(
+    bbData,
+    "buildBaron.buildBaronConfigured",
+    undefined
```
Comment (khelif96) [inline src/analytics/task/useAnnotationAnalytics.ts:64]: could we use `eventData?.task?.annotation` instead we wanted to move away from _.get

**Review by khelif96 — APPROVED** [review]
